mlir/tools/miopen-gen/miopen-gen.cpp

```diff
@@ -690,141 +690,31 @@ static FuncOp createGPUWrapper(ModuleOp &module, const KernelIF &kernel) {
-    if (elemType.isF32()) {
-    } else if (elemType.isF16()) {
-      allocFuncName = "mgpuMemAlloc5DHalf";
-    } else if (elemType.isBF16()) {
-      allocFuncName = "mgpuMemAlloc5DBF16";
-    } else if (elemType.isInteger(8)) {
-      allocFuncName = "mgpuMemAlloc5DInt8";
-    } else if (elemType.isInteger(32)) {
-      allocFuncName = "mgpuMemAlloc5DInt32";
-    }
-    auto unk5DType = MemRefType::get({-1, -1, -1, -1, -1}, elemType);
-    return makeFuncDecl(module, allocFuncName, {unk5DType}, {unk5DType});
-  };
-
-  auto getGpuDeallocFunc = [&](mlir::Type elemType) -> FuncOp {
-    StringRef deallocFuncName = "mgpuMemDealloc5DFloat";
-    if (elemType.isF32()) {
-    } else if (elemType.isF16()) {
-      deallocFuncName = "mgpuMemDealloc5DHalf";
-    } else if (elemType.isBF16()) {
-      deallocFuncName = "mgpuMemDealloc5DBF16";
-    } else if (elemType.isInteger(8)) {
-      deallocFuncName = "mgpuMemDealloc5DInt8";
-    } else if (elemType.isInteger(32)) {
-      deallocFuncName = "mgpuMemDealloc5DInt32";
-    }
-    auto unk5DType = MemRefType::get({-1, -1, -1, -1, -1}, elemType);
-    return makeFuncDecl(module, deallocFuncName, {unk5DType});
-  };
-
-  auto getGpuCopyFunc = [&](mlir::Type elemType) -> FuncOp {
-    StringRef allocFuncName = "mgpuMemCopy5DFloat";
-    if (elemType.isF32()) {
-    } else if (elemType.isF16()) {
-      allocFuncName = "mgpuMemCopy5DHalf";
-    } else if (elemType.isBF16()) {
-      allocFuncName = "mgpuMemCopy5DBF16";
-    } else if (elemType.isInteger(8)) {
-      allocFuncName = "mgpuMemCopy5DInt8";
-    } else if (elemType.isInteger(32)) {
-      allocFuncName = "mgpuMemCopy5DInt32";
-    }
-    auto unk5DType = MemRefType::get({-1, -1, -1, -1, -1}, elemType);
-    return makeFuncDecl(module, allocFuncName,
-                        {unk5DType, unk5DType, b.getIntegerType(32)});
-  };
-
   SmallVector<mlir::Value, 4> cpuMem;
   SmallVector<mlir::Value, 4> gpuMem;
-  SmallVector<mlir::Value, 4> paramCasts;
-  uint32_t i = 0;
-  for (auto &paramType : kernel.params) {
-    auto shapedType = paramType.dyn_cast<ShapedType>();
-    assert(shapedType);
-
-    auto elemType = shapedType.getElementType();
-
-    // Emit memref.expand_shape (if needed)
-    auto fiveDimUnknownSizeMemRefType =
-        MemRefType::get({-1, -1, -1, -1, -1}, elemType);
-
-    // Emit memref.cast.
-    mlir::Value arg = block->getArgument(i++);
-    mlir::Value castOp = b.create<memref::CastOp>(
-        loc, fiveDimUnknownSizeMemRefType, makeNDMemRef(b, arg, 5));
-    cpuMem.push_back(castOp);
+  for (auto pair : llvm::enumerate(kernel.params)) {
+    mlir::Value arg = block->getArgument(pair.index());
+    cpuMem.push_back(arg);
 
     // Emit GPU memory allocation function calls.
-    auto gpuAllocOp = b.create<func::CallOp>(loc, getGpuAllocFunc(elemType),
-                                             ValueRange{castOp});
+    auto gpuAllocOp =
+        b.create<gpu::AllocOp>(loc, arg.getType(), mlir::Type(), ValueRange{},
```

Comment:
Would it be possible to give those default-valued arguments (the `ValueRange{}` and the like) a `/*parameterName=*/` comment so we know what they're meant to be? Also, are they required arguments?

(Note: this isn't at all blocking, feel free to ignore this suggestion and merge)

Reply:
looks like they are not required if alloc is not called in the async manner.